types/dec_coin.go

```diff
@@ -60,6 +58,39 @@ func NewDecCoinFromCoin(coin Coin) DecCoin {
 	}
 }
 
+// NewInt64DecCoin returns a new DecCoin with a denomination and amount. It will
+// panic if the amount is negative or denom is invalid.
+func NewInt64DecCoin(denom string, amount int64) DecCoin {
+	return NewDecCoin(denom, NewInt(amount))
+}
+
+// SameDenomAs returns true if the two DecCoin have the same denom.
+func (coin DecCoin) SameDenomAs(other DecCoin) bool {
+	return (coin.Denom == other.Denom)
+}
+
+// IsZero returns if the DecCoin amount is zero.
+func (coin DecCoin) IsZero() bool {
+	return coin.Amount.IsZero()
+}
+
+// IsGTE returns true if they are the same type and the receiver is
+// an equal or greater value.
+func (coin DecCoin) IsGTE(other DecCoin) bool {
+	return coin.SameDenomAs(other) && (!coin.Amount.LT(other.Amount))
```

Comment:
```suggestion
	return coin.SameDenomAs(other) && !coin.Amount.LT(other.Amount)
```

Reply:
also, I find the returning false on different denom convention confusing, can we panic instead?

Reply:
Hmmm, perhaps we should panic. On which methods should we panic? Any logic or arithmetic-based methods that compare two different coins?

Reply:
As far as I can tell this function is never used, maybe we should just delete it

Reply:
It's not used, correct, but maybe other upstream clients are using it? Don't we want to cover all basic logic and arithmetic cases for coin/s?

Reply:
Personally I think it's better for the caller to just compare amounts directly, and validate the denom themselves, I just don't think this function adds much.

Weak opinion though, otherwise panicking on different denoms seems best

Reply:
Fair enough -- will panic on different denoms.